src/pfe/file-watcher/server/src/projects/projectUtil.ts

```diff
@@ -401,10 +401,12 @@ async function executeBuildScript(operation: Operation, script: string, args: Ar
 
                     logger.logProjectInfo("The project location for " + operation.projectInfo.projectID + " is " + projectLocation, projectID, projectName);
 
+                    // this if loop is used to get the odo app name and set the odo app name in the project info file and reload it back
                     if (operation.projectInfo.projectType == "odo") {
-                        // this if loop is  used to get the odo app name to set the odo app name in the project info file and reload it back
-                        await setOdoAppName(operation.projectInfo);
-                        operation.projectInfo = await getProjectInfo(operation.projectInfo.projectID);
+                        if (!operation.projectInfo.odoAppName) {
```

Comment:
I would have liked it, if the if conditions were  in a single line like`if(operation.projectInfo.projectType == "odo" && !operation.projectInfo.odoAppName)` instead of two if's. But this works as well.